### PR TITLE
Dlg histogram and dlgPlot desable options in multiple variables case ++ dlgBoxPlot solve reset bug.

### PR DESCRIPTION
**Pull request by FrancoisJRenaud**
This is about #1874 and  #2086  and error commented in #2085.

### CODE REVIEW

File: instat/dlgHistogram.vb [hunk 1]
```diff
@@ -216,6 +218,13 @@ Public Class dlgHistogram
         Else
             clsRaesFunction.RemoveParameterByName("x")
         End If
+        If ucrVariablesAsFactorforHist.bSingleVariable Then
+            cmdHistogramOptions.Enabled = True
+            cmdOptions.Enabled = True
+        Else
+            cmdHistogramOptions.Enabled = False
+            cmdOptions.Enabled = False
+        End If
```
Comment (dannyparsons) [inline instat/dlgHistogram.vb]: I would prefer if you put this in its own sub and then call it from `SetDefaults` as well as here. I don't like repeated lines and just having the `Enabled = True` at the top without checking the control, even though it should be in single case at that point.

**Comment by FrancoisJRenaud** [comment]
Should be ready to merge. Tested on all three dialogues.

File: instat/dlgPlot.vb [hunk 2]
```diff
@@ -83,7 +84,8 @@ Public Class dlgPlot
     End Sub
 
     Private Sub TestOkEnabled()
-        If (ucrReceiverX.IsEmpty() AndAlso ucrVariablesAsFactorForLinePlot.IsEmpty()) OrElse (ucrSaveLinePlot.chkSaveGraph.Checked AndAlso ucrSaveLinePlot.ucrInputGraphName.IsEmpty) Then
+        'Both x and y aesthetics are mandatory for geom_line.
```
Comment (dannyparsons) [inline instat/dlgPlot.vb]: This should be the same as point, if they are not there add = ""

Comment (FrancoisJRenaud) [inline instat/dlgPlot.vb]: Ok, so the behaviour is similar. Now this is quite a substantial edit. I should do it on another pull request. Also will have to edit ucrgeom,... If you merge this I can start new pull request. If you don't want to make these changes for the beta. I think it's safer to keep both aes mandatory. Also need to write that new method for partially mandatory aes so...

Comment (dannyparsons) [inline instat/dlgPlot.vb]: Sounds good. I suggest partially mandatory waits until after the beta, as the changes are big. For now you can make line a special case as the others are.

**Pull request merged**